pkg/controllers/collaset/synccontrol/update.go

```diff
@@ -416,3 +470,60 @@ func (u *recreatePodUpdater) GetPodUpdateFinishStatus(podInfo *PodUpdateInfo) (f
 	// Recreate policy alway treat Pod as update finished
 	return podInfo.IsUpdatedRevision && !podInfo.PodDecorationChanged, "", nil
 }
+
+type replaceUpdatePodUpdater struct {
+	collaSet *appsv1alpha1.CollaSet
+	ctx      context.Context
+	client.Client
+}
+
+func (u *replaceUpdatePodUpdater) AnalyseAndGetUpdatedPod(updatedRevision *appsv1.ControllerRevision, podUpdateInfo *PodUpdateInfo) (inPlaceUpdateSupport bool, onlyMetadataChanged bool, updatedPod *corev1.Pod, err error) {
+	// when replaceUpdate, inPlaceUpdateSupport and onlyMetadataChanged always false
+
+	// 1. judge replace pair new pod is updated revision, if not, delete.
+	if podUpdateInfo.replacePairNewPodInfo != nil {
+		newPodRevision, exist := podUpdateInfo.replacePairNewPodInfo.Pod.Labels[appsv1.ControllerRevisionHashLabelKey]
+		if exist && newPodRevision == updatedRevision.Name {
+			return
+		}
+		u.Delete(u.ctx, podUpdateInfo.replacePairNewPodInfo.Pod)
```

Comment:
Please use label `PodDeletionIndicationLabelKey` to delete the Pod in case that the Pod is not in PodOpsLifecycle.

Besides, we should consider the err returned by `delete`.